Say/ViewController.swift

```diff
@@ -90,7 +93,29 @@ class ViewController: NSViewController {
             }
         }
     }
-
+    
+    @IBAction func selectText(_ sender: NSTextField){
+        let URLString = sender.stringValue
+        let url = URL(string: sender.stringValue)!
+        var data: NSData? = nil
+        data=NSData.init(contentsOf: url)//NSurlData
+        let dataString: String = String(data:data! as Data, encoding:String.Encoding.utf8)!
+        
+        let regex = try! NSRegularExpression(pattern: "<title>\\s*(.*)\\s*</title>", options: NSRegularExpression.Options())
+    
+        let result = regex.matches(in: dataString as String, options: NSRegularExpression.MatchingOptions(), range: NSRange(location: 0, length: dataString.lengthOfBytes(using: String.Encoding.utf8)))
+        
+        if result.count > 0 {
+            let range = result[0].rangeAt(1)
+            let text = (dataString as NSString).substring(with: range)
+            textView.string=text
+            print(text)
+            
+        }
+        
+    }
+    
```

Comment:
I suggest you to use optional binding. Because there's no way to catch nil data in line 99.
`let url = URL(string: sender.stringValue)!`
If user inputs some wrong format URL, It excutes error.
I tryed changing your code, but I am beginner in Swift too... so it's not perfect.
Have a nice day! 
```
@IBAction func selectText(_ sender: NSTextField) {
    if let url = URL(string: sender.stringValue) {
        // if URL format was right
        let data = NSData.init(contentsOf: url) as! Data
        let dataString = String(data:data, encoding:String.Encoding.utf8)!
        if let result = findTitle(in: dataString) {
            textView.string = result
            print(result)
        } else {
            print("title not found")
        }
    }
}

func findTitle(in dataString: String) -> String? {
    let regex = try! NSRegularExpression(pattern: "<title>\\s*(.*)\\s*</title>", options: NSRegularExpression.Options())
    let result = regex.matches(in: dataString as String, options: NSRegularExpression.MatchingOptions(), range: NSRange(location: 0, length: dataString.lengthOfBytes(using: String.Encoding.utf8)))
    if result.count > 0 {
        let range = result[0].rangeAt(1)
        let text = (dataString as NSString).substring(with: range)

        return text
    } else {
        return nil
    }
}
```